iothub/service/samples/how to guides/CleanupDevicesSample/CleanupDevicesSample.cs

```diff
@@ -93,31 +93,31 @@ private async Task CleanupDevices(int deviceCount)
                     } while (jobTimer.Elapsed < s_maxJobDuration);
 
                     // Wait until job is finished.
-                    IotHubJobResponse jobToBeDeleted = null;
                     jobTimer.Restart();
+                    IotHubJobResponse deleteDevicesJob = null;
                     while (jobTimer.Elapsed < s_maxJobDuration)
                     {
-                        jobToBeDeleted = await _hubClient.Devices.GetJobAsync(importDevicesToBeDeletedJob.JobId);
```

Comment:
This was the bug. The value we got back from the method wasn't the one we checked for status on 101 and later.